client/src/main/java/org/apache/rocketmq/client/impl/consumer/RebalanceImpl.java

```diff
@@ -278,11 +278,20 @@ private void rebalanceByTopic(final String topic, final boolean isOrder) {
 
                     List<MessageQueue> allocateResult = null;
                     try {
-                        allocateResult = strategy.allocate(
-                            this.consumerGroup,
-                            this.mQClientFactory.getClientId(),
-                            mqAll,
-                            cidAll);
+                        if (!isRebalanceByBroker()) {
+                            allocateResult = strategy.allocate(
+                                this.consumerGroup,
+                                this.mQClientFactory.getClientId(),
+                                mqAll,
+                                cidAll);
```

Comment:
IMO, It would be better that the client only needs to bring the current cid to get the result. 
Both mqAll and cidAll are obtained on the broker side.

Reply:
Updated. According to our previous discussion, the performance of obtaining mqAll from broker side still need further discussions.